homeassistant/components/devolo_home_control/sensor.py

```diff
@@ -0,0 +1,92 @@
+"""Platform for sensor integration."""
+import logging
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+from .devolo_device import DevoloDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all sensor devices and setup them via config entry."""
+    entities = []
+
+    for device in hass.data[DOMAIN]["homecontrol"].multi_level_sensor_devices:
+        for multi_level_sensor in device.multi_level_sensor_property:
+            entities.append(
+                DevoloMultiLevelDeviceEntity(
+                    homecontrol=hass.data[DOMAIN]["homecontrol"],
+                    device_instance=device,
+                    element_uid=multi_level_sensor,
+                )
+            )
+    async_add_entities(entities, False)
+
+
+class DevoloMultiLevelDeviceEntity(DevoloDeviceEntity):
+    """Representation o a multi level sensor within devolo Home Control."""
+
+    def __init__(self, homecontrol, device_instance, element_uid):
+        """Initialize a devolo multi level sensor."""
+        super().__init__(
+            homecontrol=homecontrol,
+            device_instance=device_instance,
+            element_uid=element_uid,
+            name=f"{device_instance.itemName} {device_instance.multi_level_sensor_property.get(element_uid).sensor_type}",
+            sync=self._sync,
+        )
+
+        self._state = self._device_instance.multi_level_sensor_property.get(
+            element_uid
+        ).value
+
+        self._multi_level_sensor_property = device_instance.multi_level_sensor_property.get(
+            element_uid
+        )
+        if self._multi_level_sensor_property.sensor_type == "temperature":
```

Comment:
Instead of an if/else tree, this could be a dictionary to provide the mapping.
e.g.,:

```python
DEVICE_CLASS_MAPPING = {
    "temperature": DEVICE_CLASS_TEMPERATURE,
    "light": DEVICE_CLASS_ILLUMINANCE,
    "humidity": DEVICE_CLASS_HUMIDITY,
}
```

Next, simply look it up:

```python
self._device_class = DEVICE_CLASS_MAPPING.get(self._multi_level_sensor_property.sensor_type)
```

Reply:
Done in 85988d06b2979e56dd4250e1a922f3bb7bb1aee7